crates/oxc_linter/src/lint_adapter.rs

```diff
@@ -0,0 +1,1208 @@
+        let quasi = &tlit.quasis[0].value;
+        FieldValue::String(quasi.cooked.as_ref().unwrap_or_else(|| &quasi.raw).as_str().into())
+    } else {
+        FieldValue::Null
+    }
+}
+
+fn assignment_target_to_vertex<'a>(at: &'a AssignmentTarget<'a>) -> Vertex<'a> {
+    match &at {
+        AssignmentTarget::SimpleAssignmentTarget(simple_assignment_target) => {
+            match simple_assignment_target {
+                SimpleAssignmentTarget::AssignmentTargetIdentifier(ident) => {
+                    Vertex::AssignmentToIdentifier(&ident.span, &ident.name)
+                }
+                _ => Vertex::AssignmentOnObject(&simple_assignment_target),
+            }
+        }
+        AssignmentTarget::AssignmentTargetPattern(d) => {
+            Vertex::DestructuringAssignment(DestructuringAssignment::AssignmentExpression(d))
+        }
+    }
+}
+
+// impl From<&AssignmentTargetPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &AssignmentTargetPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::AssignmentExpression(ap))
+//     }
+// }
+
+// impl From<&ArrayPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &ArrayPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::ArrayPattern(ap))
+//     }
+// }
+
+// impl From<&ObjectPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &ObjectPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::ObjectPattern(ap))
+//     }
+// }
+
+// impl From<&AssignmentPattern<'_>> for Vertex<'_> {
+//     fn from(ap: &AssignmentPattern) -> Self {
+//         Vertex::DestructuringAssignment(DestructuringAssignment::DefaultedDestructuringAssignment(
+//             ap,
+//         ))
+//     }
+// }
+
+fn binding_pattern_kind_to_vertex<'a>(bpk: &'a BindingPatternKind<'a>) -> Vertex<'a> {
+    match bpk {
+        BindingPatternKind::BindingIdentifier(id) => {
+            Vertex::AssignmentToIdentifier(&id.span, &id.name)
+        }
+        BindingPatternKind::ArrayPattern(ap) => {
+            Vertex::DestructuringAssignment(DestructuringAssignment::ArrayPattern(ap))
+        }
+        BindingPatternKind::ObjectPattern(op) => {
+            Vertex::DestructuringAssignment(DestructuringAssignment::ObjectPattern(op))
+        }
+        BindingPatternKind::AssignmentPattern(ap) => Vertex::DestructuringAssignment(
+            DestructuringAssignment::DefaultedDestructuringAssignment(ap),
+        ),
+    }
+}
+
+impl<'a> Vertex<'a> {
+    fn ts_type_annotation_coercion(&self) -> bool {
+        match self {
+            Vertex::TSTypeAnnotation(_) => true,
+            Vertex::AstNode(ast) => match ast.kind() {
+                AstKind::TSTypeAnnotation(_) => true,
+                _ => false,
+            },
+            _ => false,
+        }
+    }
+
+    fn ts_type_annotation_type_annotation(&self) -> Box<dyn Iterator<Item = Vertex<'a>> + 'a> {
```

Comment:
```suggestion
    fn ts_type_annotation_type_annotation(&self) -> VertexIterator<'a, Vertex<'a>> {
```
Possibly a bit easier on the eyes.

Also appears in a few other places.

Reply:
Fixed, thanks!